File: src/main/java/com/github/zxh/classpy/classfile/bytecode/InvokeInterface.java
package com.github.zxh.classpy.classfile.bytecode;

import com.github.zxh.classpy.classfile.constant.ConstantPool;
import com.github.zxh.classpy.classfile.jvm.Opcode;

public class InvokeInterface extends Instruction {

    {
        u1  ("opcode");
        u2cp("index");
        u1  ("count");
        u1  ("zero");
    }

    public InvokeInterface(Opcode opcode, int pc) {
        super(opcode, pc);
    }
    
    @Override
    protected void postRead(ConstantPool cp) {
        setDesc(getDesc() + " "
                + super.get("index").getDesc() + ", "
                + super.getUInt("count"));
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/bytecode/InvokeInterface.kt
package com.github.zxh.classpy.classfile.bytecode

import com.github.zxh.classpy.classfile.constant.ConstantPool

class InvokeInterface(opcode: Opcode, pc: Int) : Instruction(opcode, pc) {

    init {
        u1  ("opcode")
        u2cp("index")
        u1  ("count")
        u1  ("zero")
    }

    override fun afterRead(cp: ConstantPool) {
        val index = super.get("index").desc
        val count = super.getInt("count")
        desc = "${opcode.name} $index, $count"
    }
    
}
